com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/LaunchRequestHandler.java

```diff
@@ -87,6 +97,62 @@
 
         activeLaunchHandler.preLaunch(launchArguments, context);
 
+        Path tempfile = null;
```

Comment:
The body of this function is already long. Please refactor by putting the procedures of generating Jar Manifest and ArgFile into different utility functions. Those functions should be functional without referencing states from other parts. After that the body looks like:

```java
if (launchArguments.shortenCommandLine == ShortenApproach.JARMANIFEST) {
    Path path = AdapterUtils.generateClasspathJar(...);
    context.setClasspathJar(path);
} else if (...) {
}
```